### ISSUE TITLE
Invalid coverage report

### ISSUE THREAD

**Issue by kevinkasper** [analogjs/analog#1212]
### Please provide the environment you discovered this bug in.

Clone minimal reproducible example:

https://github.com/kevinkasper/analogjs-vite-plugin-angular-coverage-mre

1. `yarn install`
2. `yarn vitest --coverage --run`

### Which area/package is the issue in?

vite-plugin-angular

### Description

If the minimal reproducible example is executed, a code coverage of 100% is displayed, although no test covers the code.

Many thanks in advance!

### Please provide the exception or error you saw

```true
❯ yarn vitest --coverage --run
yarn run v1.22.22
$ /Users/FooBar/analogjs-vite-plugin-angular-coverage-mre/node_modules/.bin/vitest --coverage --run

 RUN  v2.0.3 /Users/FooBar/analogjs-vite-plugin-angular-coverage-mre
      Coverage enabled with v8

 ✓ src/app/app.component.spec.ts (1)
   ✓ AppComponent (1)
     ✓ test

 Test Files  1 passed (1)
      Tests  1 passed (1)
   Start at  22:48:35
   Duration  793ms (transform 216ms, setup 161ms, collect 187ms, tests 34ms, environment 152ms, prepare 216ms)

 % Coverage report from v8
------------------|---------|----------|---------|---------|-------------------
File              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------|---------|----------|---------|---------|-------------------
All files         |     100 |      100 |     100 |     100 |                   
 app.component.ts |     100 |      100 |     100 |     100 |                   
------------------|---------|----------|---------|---------|-------------------
✨  Done in 2.42s.
```

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [X] No

**Comment by DerHerrGammler** [comment]
Same issue not only with `v8` coverage provider but as well with `istanbul` coverage provider.

**Comment by AlexAegis** [comment]
This is a blocking issue for us to adopt vitest as our testing framework. It runs nice and a lot faster but precise coverage is a must

**Comment by brandonroberts** [comment]
Fix for coverage reports has landed in the `1.9.0-beta.12` release.  I've done some testing but please validate the fix on your end also
